ssntp/client.go

```diff
@@ -77,6 +77,7 @@ type Client struct {
 	closed    chan struct{}
 
 	frameWg              sync.WaitGroup
+	frameWgMutex         sync.Mutex
```

Comment:
This one is odd as well.  You shouldn't need to protect a WaitGroup with a mutex.  By design they are expected to be called safely from different go routines.   Could you paste the race detector logs?